p2p/src/peer_manager/mod.rs

```diff
@@ -144,6 +144,9 @@ where
     subscribed_to_peer_addresses: BTreeSet<PeerId>,
 
     peer_eviction_random_state: peers_eviction::RandomState,
+
+    // Map of transactions for which we need to request a body.
+    pending_transaction_requests: Mutex<BTreeMap<Id<Transaction>, OneshotQueue<PeerId>>>,
```

Comment:
Why is `Mutex` used here? `PeerManager` is single-threaded.

Reply:
Good catch! Removed. I originally played around with a different pattern that warranted a mutex where we would spawn transaction trackers. However, I decided to drop the idea due to complexity + it probably scales worse.